core/src/main/java/tc/oc/pgm/kits/TeamSwitchKit.java

```
@@ -0,0 +1,23 @@
+package tc.oc.pgm.kits;
+
+import tc.oc.pgm.api.player.MatchPlayer;
+import tc.oc.pgm.teams.TeamFactory;
+import tc.oc.pgm.teams.TeamMatchModule;
+
+public class TeamSwitchKit extends DelayedKit {
+
+  private final TeamFactory team;
+
+  public TeamSwitchKit(TeamFactory team) {
+    this.team = team;
+  }
+
+  @Override
+  public void applyDelayed(MatchPlayer player, boolean force) {
+    TeamMatchModule tmm = player.getMatch().getModule(TeamMatchModule.class);
```

i'd just require team match module, there's no point in silently ignoring when someone tries to use this in FFA

---

Great idea! Will adjust 👍